Project MUSE.js

```diff
@@ -1,182 +1,116 @@
-				if(link && link.href && title && title.textContent) {
-					items[link.href] = title.textContent;
-				}
-			}
-		}
 
-		Zotero.selectItems(items, function(items) {
-			if(!items) {
-				return true;
-			}	
-			var i;
-			var urls = [];
-			for (i in items) {urls.push(i);};
-			Zotero.Utilities.processDocuments(urls, scrapeOne, function() {Zotero.done();}, null);		
-		});
-	} else scrapeOne(doc);
-	Zotero.wait();
+function doWeb(doc, url) {
+    if (detectWeb(doc, url) == "multiple") {
+        Zotero.selectItems(getSearchResults(doc, false), function (items) {
+            if (!items) {
+                return true;
+            }
+            var articles = [];
+            for (var i in items) {
+                articles.push(i);
+            }
+            ZU.processDocuments(articles, scrape);
+        });
+    } else {
+        scrape(doc, url);
+    }
 }
 
-// Given an article page, get the RIS and open it
-function scrapeOne(doc) {
-	var url = doc.location.href;
-	var hostRe = new RegExp("^(https?://[^/]+)/");
-		var m = hostRe.exec(url);
-		var host = m[1];
+function scrape(doc, url) {
+	var abstract = ZU.xpathText(doc, '//div[@class="abstract"]');
+	if (!abstract) abstract = ZU.xpathText(doc, '//div[@class="description"][1]/text()');
+	var translator = Zotero.loadTranslator('web');
+    // Embedded Metadata
+    translator.setTranslator('951c027d-74ac-47d4-a107-9c3069ab7b48');
+    translator.setDocument(doc);
+    translator.setHandler('itemDone', function (obj, item) {
+        if (abstract) {
+        	item.abstractNote = abstract.replace(/^\s*Abstract/, "").replace(/,\s*$/, "").trim();
+        }
+        if (url.indexOf("/article/") != -1) {
+       		var pdfurl = url.replace(/(\/article\/\d+).*/, "$1") + "/pdf";
+       		Z.debug(pdfurl)
+       		item.attachments = [{
+						"url": pdfurl,
+						"title": "Full Text PDF",
+						"mimeType": "application/pdf"
+					}]
+        }
+        item.libraryCatalog = "Project MUSE";
+        item.complete();
+    });
 
-	if (url.match(/\/books\//)){
-		var abstract = ZU.xpathText(doc, '//div[@class="book_info"]/div[@class="description"]')
-		var citurl = ZU.xpathText(doc, '//li[@class="view_citation"]/a/@href');
-		var newUrl = host + citurl;
-			Zotero.Utilities.HTTP.doGet(newUrl, function(text) {
-			text=text.match(/TY  -(.+\n)+ER  -/)[0]
-				var translator = Zotero.loadTranslator("import");
-			//set RIS translator
-			translator.setTranslator("32d59d2d-b65a-4da4-b0a3-bdd3cfb979e7");
-			translator.setString(text);
-			translator.setHandler("itemDone", function(obj, item) {
-			item.attachments.push({document:doc, title:"Project MUSE Snapshot"});
-			if (abstract){
-				item.abstractNote=abstract.replace(/\n/g, " ").replace(/\s\s+/g, " ");
-			}
-			item.complete();
-			});
-			translator.translate();
-			})
-	}
-	
-	else{
+    translator.getTranslatorObject(function(trans) {
+        trans.doWeb(doc, url);
+    });
```

Comment:
If you don't need any special handling here, then you can omit this function completely.